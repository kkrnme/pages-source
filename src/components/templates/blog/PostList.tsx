import React from "react"
import LinkToPost from "../../general/LinkToPost"
import { Mdx, MdxEdge } from "../../../../types/graphqlTypes"

export const PostList = ({ edges: posts }: PostListProps) => (
  <ul className="list-none p-0 sm:flex-wrap sm:flex ">
    {posts.map(({ node }) => (
      <li className="w-full sm:w-1/2 lg:w-1/3 p-1" key={node.id ?? undefined}>
        <LinkToPost
          to={node.frontmatter?.path!}
          title={node.frontmatter?.title!}
          excerpt={node.excerpt}
          status={node.frontmatter?.status!}
        />
      </li>
    ))}
  </ul>
)

export type PostListProps = {
  edges: {
    node: Pick<Mdx, "excerpt" | "id" | "frontmatter">
  }[]
}

export default PostList
