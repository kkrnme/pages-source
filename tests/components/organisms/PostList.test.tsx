import React from "react"
import renderer from "react-test-renderer"

import PostList from "../../../src/components/templates/blog/PostList"

describe("PostList", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <PostList
          {...{
            posts: [
              {
                excerpt:
                  "Lorem ipsum dolor, sit amet consectetur adipisicing elit. ",
                id: `c3884f97-b0d1-8e34-edc7-af4a-3ce9fe72e66`,
                path: "hoge",
                title: "Lorem ipsum",
              },
            ],
          }}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

const random16 = () => Math.round(Math.random() * 0xffff).toString(16)
