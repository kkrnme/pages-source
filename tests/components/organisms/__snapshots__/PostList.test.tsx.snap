// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList renders correctly 1`] = `
<ul
  className="css-1a6qcq1"
>
  <li>
    <a
      className="css-1yrbkum"
      href="/hoge/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div>
        <h2>
          <span>
            Lorem ipsum
          </span>
        </h2>
      </div>
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. 
      </p>
    </a>
  </li>
</ul>
`;
